app/workers/deletes.php

```diff
@@ -319,22 +324,28 @@ protected function deleteForProjectIds(callable $callback): void
 
         $executionStart = \microtime(true);
 
-        while($sum === $limit) {
+        while ($sum === $limit) {
             Authorization::disable();
             $projects = $this->getConsoleDB()->find('projects', [], $limit, ($chunk * $limit));
             Authorization::reset();
 
             $chunk++;
 
-            $projectIds = array_map (function ($project) { 
-                return $project->getId(); 
-            }, $projects);
+            /** @var string[] $projectIds */
+            $projectIds = array_map(fn(Document $project) => $project->getId(), $projects);
 
             $sum = count($projects);
 
             Console::info('Executing delete function for chunk #' . $chunk . '. Found ' . $sum . ' projects');
             foreach ($projectIds as $projectId) {
-                $callback($projectId);
+                if (!($dbForInternal = $this->getInternalDB($projectId))) {
+                    throw new Exception('Failed to get projectDB for project ' . $projectId);
+                }
+
+                if (!($dbForExternal = $this->getExternalDB($projectId))) {
+                    throw new Exception('Failed to get projectDB for project ' . $projectId);
+                }
+                $callback($projectId, $dbForInternal, $dbForExternal);
```

Comment:
I can understand that this change makes the code a bit more DRY, but I got totally lost when trying to follow the database connections through the logic. Maybe we should repeat ourselves a bit to make the story easier to follow?

Reply:
Actually, the underlying method already throws an exception - so I can remove that completely from the Deletes class 👍🏻